### DISCUSSION TITLE
Error : Device platform (ios) has no napalm driver

### DISCUSSION THREAD

**Discussion by alensfor** [DanSheps/netbox-config-backup#97] [Q&A] [answered]
Hello all,
I am not sure if I missed a bit of documentation, I am struggling to configure this plugin in my environment.
When adding a backup and a single device, I get this error "Device platform (ios) has no napalm driver " for all type of platforms I have (ios, junos).

I am running Netbox 4.0.3
My python venv reports the pip packages are installed.
```
$ pip3 freeze | grep napalm
napalm==4.1.0
netbox-napalm-plugin==0.2.1
```
I have created a new permission `napalm_read` and my user received this permission.
The plugin is netbox configured with credentials (user/pwd).

I tested napalm from the venv successfully on the same device, with :
```python
### test napalm
from napalm import get_network_driver
import getpass
import json
from tabulate import tabulate

passwd = getpass.getpass('Please enter the password: ') # Reads the output from the user and save it as a string

driver = get_network_driver('ios')
sealabedgrtr01 = {
    "hostname": 'edited', 
    "username": "edited",
    "password": passwd,
    "optional_args": {"secret": passwd}
    }

sealabedgrtr01_conn = driver(**sealabedgrtr01)
sealabedgrtr01_conn.open()
print(f"Connecting to {sealabedgrtr01['hostname']}")
output = sealabedgrtr01_conn.get_mac_address_table() #get mac-address table method
output_json = json.dumps(output, indent=4)
print(output_json)
```

From what I can read, we no longer need to install pip packages like `napalm-ios` etc. (anyway, the script above is successful)

My device in Netbox is configured with a platform = ios
And with a primary/management IP.

I could use your help to understand what I failed to do or what I could do to get more details regarding the cause for this error.
Thank you all !

**Comment by DanSheps** [comment]
Have you configured the NetBox Napalm platform setting?

**Reply by alensfor** [reply]
Hi Dan,
I forgot to add my screenshots.
From what I understand, this is a deprecated setting, no longer existing in Netbox 4.x, the platforms configured in Netbox no longer have a Napalm driver.
Should I add this as a custom field in netbox for my platforms ?
<img width="1133" alt="Screenshot at Jul 09 17-25-04" src="https://github.com/DanSheps/netbox-config-backup/assets/67714495/29757dce-0f75-45f3-a14a-5139a2e9061d">
<img width="1177" alt="Screenshot at Jul 09 17-00-48" src="https://github.com/DanSheps/netbox-config-backup/assets/67714495/6f1c4809-49e1-45e6-8c8f-b1da7bd5d358">

**Reply by DanSheps** [reply, marked as answer]
![image](https://github.com/DanSheps/netbox-config-backup/assets/11049792/402ce1f6-048d-486c-a0b9-df54f0cbce43)
![image](https://github.com/DanSheps/netbox-config-backup/assets/11049792/a395c43d-411a-463d-90f6-c937ce3ba69f)
![image](https://github.com/DanSheps/netbox-config-backup/assets/11049792/c65283db-ca4e-43d6-8510-e69a5ad4b84c)

**Comment by alensfor** [comment]
Thank you Dan, my bad, I completely skipped this part.
It is now working perfectly ! I'm loving this plugin.

If that is ok I'd like to ask a few more things :
1. This is for me to find a way to automatically push the local git repo to the remote origin ? (or maybe this is already planned for the future ?)
2. Is there also future plans to allow to do searches across the configuration saved (latest and old configurations) ?

Thanks again
